lib/foreman_rh_cloud.rb

```diff
@@ -3,17 +3,27 @@
 require 'uri'
 
 module ForemanRhCloud
+  def self.on_premise_url
+    return unless ForemanRhCloud.with_local_advisor_engine?
+    port = ENV['ADVISOR_ENGINE_PORT'] || "8000"
+    ENV['ADVISOR_ENGINE_URL'] || "http://#{ForemanRhCloud.foreman_host.fqdn || 'localhost'}:#{port}"
```

Comment:
Tested it. Https works. So lets add that

```suggestion
    ENV['ADVISOR_ENGINE_URL'] || "https://#{ForemanRhCloud.foreman_host.fqdn || 'localhost'}:#{port}"
```

Reply:
FWIW, the current deployment will only ever work for `localhost` with HTTPS, as that's what https://github.com/theforeman/puppet-certs/pull/474 generates.

Reply:
hmm  both seem to give me the same response.

```bash
$ curl  https://buku.paji.example.com:8444//r/insights/v1/static/release/insights-core.egg
curl: (35) error:0A0000C6:SSL routines::packet length too long
[vagrant@buku ~]$ curl  https://localhost:8444//r/insights/v1/static/release/insights-core.egg
curl: (35) error:0A0000C6:SSL routines::packet length too long
```

Reply:
is your engine running with ssl enabled right now? (it only does so when you give it certs)

Reply:
while 
```bash
curl  http://localhost:8444/r/insights/v1/static/release/insights-core.egg > /dev/null
 % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 1304k  100 1304k    0     0  42.4M      0 --:--:-- --:--:-- --:--:-- 41.0M
```
seems to work 

I think we need the mtls patch for the engine

Reply:
no, you need to boot your engine with certs provided, then it will only work with https :)

Reply:
Lets do this https change in another PR. I need more time to test with the installer. I think a separate PR is good idea. We'll leave it as http for now